### DISCUSSION TITLE
Using Special K in lazer allowed?

### DISCUSSION THREAD

**Discussion by Matthias-VdC** [ppy/osu#29607] [Q&A] [answered]
Would using a program like [Special K](https://www.special-k.info/) to try to reduce latency / improve frame pacing be allowed in lazer?

**Comment by peppy** [comment, marked as answer]
hard to give a definitive answer for apps liek this. it could cause random shit to happen which could lead to getting you banned. hard to say. i'd proably test thoroughly offline to know what you're getting yourself into before using it for online play, at very least.

**Reply by 2husecondary** [reply]
Hi peppy, I've actually been using Special K on lazer lately. Apparently you can even inject Nvidia Reflex into the game, but I haven't thoroughly confirmed that works yet. I'm not technically inclined enough, so I'd like your opinion on Special K itself, as it's an open source application and you can go through to see what it does. See https://github.com/SpecialKO/SpecialK

I wanted to know if it is against the rules, if it actually does anything useful, if it breaks anything in osu.. I've been told that the frame pacing in lazer is awful apparently, but it was just a player, not a developer that said this. I'm not knowledgable enough nor do I want to trust an LLM to tell me about this, so I figured I'd necrobump this thread and see if you'd be willing to give a more thought-out opinion on Special K and explain if anything's up with frame pacing. I'm sure you know the inner workings of lazer way more than anybody else.
